### ISSUE TITLE
Failed to parse the module configuration: [undefined method `with_settings' for nil:NilClass

### ISSUE THREAD

**Issue by jordansissel** [elastic/logstash#7346]
When playing with modules, I was unsuccessful and ran into this error:

```
% bin/logstash --modules fancypants
...

[2017-06-04T01:37:49,929][ERROR][logstash.config.sourceloader] Could not fetch all the sources {:exception=>LogStash::ConfigLoadingError, :message=>"Failed to parse the module configuration: [undefined method `with_settings' for nil:NilClass]", :backtrace=>["/home/jls/projects/logstash/logstash-core/lib/logstash/config/modules_common.rb:39:in `pipeline_configs'", "org/jruby/RubyArray.java:1613:in `each'", "/home/jls/projects/logstash/logstash-core/lib/logstash/config/modules_common.rb:25:in `pipeline_configs'", "/home/jls/projects/logstash/logstash-core/lib/logstash/config/source/modules.rb:14:in `pipeline_configs'", "/home/jls/projects/logstash/logstash-core/lib/logstash/config/source_loader.rb:59:in `fetch'", "org/jruby/RubyArray.java:2409:in `collect'", "/home/jls/projects/logstash/logstash-core/lib/logstash/config/source_loader.rb:58:in `fetch'", "/home/jls/projects/logstash/logstash-core/lib/logstash/agent.rb:146:in `converge_state_and_update'", "/home/jls/projects/logstash/logstash-core/lib/logstash/agent.rb:88:in `execute'", "/home/jls/projects/logstash/logstash-core/lib/logstash/runner.rb:333:in `execute'", "org/jruby/RubyProc.java:281:in `call'", "/home/jls/projects/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.22/lib/stud/task.rb:24:in `initialize'"]}
[2017-06-04T01:37:49,930][ERROR][logstash.agent           ] An exception happened when converging configuration {:exception=>RuntimeError, :message=>"Could not fetch the configuration, message: Failed to parse the module configuration: [undefined method `with_settings' for nil:NilClass]", :backtrace=>["/home/jls/projects/logstash/logstash-core/lib/logstash/agent.rb:153:in `converge_state_and_update'", "/home/jls/projects/logstash/logstash-core/lib/logstash/agent.rb:88:in `execute'", "/home/jls/projects/logstash/logstash-core/lib/logstash/runner.rb:333:in `execute'", "org/jruby/RubyProc.java:281:in `call'", "/home/jls/projects/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.22/lib/stud/task.rb:24:in `initialize'"]}
```

This error seems to occur when a module is given but does not exist. THe error message is a bug (undefined method `with_settings`). Ideally, this is more actionable, such as "The module `fancypants` does not exist. Here are the list of known modules: x, y, z ..."

**Comment by guyboertje** [comment]
We don't have any tests in the runner spec for this. I guess that is my first task.

**Comment by guyboertje** [comment]
Fixed with #7367
